src/main/java/com/google/sps/servlets/Caption.java

```diff
@@ -38,8 +38,13 @@ private String parseXmlFromStream(InputStream inputStream)
     NodeList textDom = doc.getElementsByTagName("text");
     StringBuilder caption = new StringBuilder();
     for (int i = 0; i < textDom.getLength(); i++) {
-      caption.append(textDom.item(i).getFirstChild().getNodeValue());
-      caption.append(" ");
+      try {
+        caption.append(textDom.item(i).getFirstChild().getNodeValue());
+        caption.append(" ");
+      } catch (NullPointerException ok) {
+        // getFirstChild() returns null if the item has no text
```

Comment:
instead of catching NPE you could consider checking for null before calling getNodeValue. something like:

```
const child = textDom.item(i).getFirstChild();
if (child) {
  caption.append(child.getNodeValue());
  caption.append(" ");
}
```

I like that approach because I feel like it removes the need for the comment in your catch block.

Reply:
Done. Thanks for the suggestion!